app/src/main/java/com/mborowiec/antennapod/activity/MainActivity.java

```diff
@@ -490,6 +495,22 @@ private void handleNavIntent() {
     protected void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
         setIntent(intent);
+
+        handleIntent(intent);
+    }
+
+    /**
+     * Checks whether the activity was launched via a deep link.
+     *
+     * @param intent incoming intent
+     */
+    private void handleIntent(Intent intent) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```

Comment:
What about changing the order to reduce the nesting level, e.g., like

if (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT) {
   return;
}

// remaining code